src/app/features/rpc-request/hooks/use-rpc-broadcast-stacks-transaction.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import type { StacksTransactionWire } from '@stacks/transactions';
+
+import type { Money } from '@leather.io/models';
+import { type RpcMethodNames, createRpcSuccessResponse } from '@leather.io/rpc';
+import { isString } from '@leather.io/utils';
+
+import { logger } from '@shared/logger';
+import { RouteUrls } from '@shared/route-urls';
+
+import { useStacksBroadcastTransaction } from '@app/common/transactions/stacks/use-stacks-broadcast-transaction';
+import { useSignStacksTransaction } from '@app/store/transactions/transaction.hooks';
+import { useUiActions } from '@app/store/ui/ui.slice';
+
+import { useRpcRequestParams } from './use-rpc-request-params';
+
+interface UserSetTxOptions {
+  fee: Money | null;
+  nonce: number | string;
+}
+
+export function useRpcBroadcastStacksTransaction(method: RpcMethodNames) {
+  const { tabId, requestId } = useRpcRequestParams();
+  const signStacksTransaction = useSignStacksTransaction();
+  const stacksBroadcastTransaction = useStacksBroadcastTransaction();
+  const { setSubmitted } = useUiActions();
+  const navigate = useNavigate();
+
+  return useCallback(
+    async (unsignedTx: StacksTransactionWire, { fee, nonce }: UserSetTxOptions) => {
+      fee && unsignedTx.setFee(fee.amount.toNumber());
```

Comment:
```suggestion
      if (fee) unsignedTx.setFee(fee.amount.toNumber());
```

outside conditional jsx, think we should stick to ifs